Compiler-Internals.md

```diff
@@ -18,19 +19,27 @@ The checker then waits until a particular symbol is requested to calculate type
 Each phase of the compiler (parser, binder, etc -- see below for details) treats data structures from the previous phases as immutable.
 In addition, data structures created within each phase are not usually modified after their creation.
 This requires a look-aside table in some cases.
-For example, because the binder only looks at one file at a time, the checker needs a merged-symbols table to track merged declarations. 
+For example, because the binder only looks at one file at a time, the checker needs a merged-symbols table to track merged declarations.
 It checks whether a symbol has an entry in the merged-symbols table each time before it uses a symbol.
 
 # Parser
 
+It's a recursive descent parser. It's pretty resilient, so if you search for functions matching the thing you want to change, you can probably get away with just adding the code to parse.
+There aren't any surprises in the general implementation style here.
+
 ## Incremental parsing
 
 ## ECMAScript parsing contexts
 
 ## ECMAScript automatic semicolon insertion
 
+## JSDoc parsing
+
 # Checker
 
+The checker is almost 20,000 lines long, and does almost everything that's not syntactic.
+Surprisingly, a checker gets created every time the language service requests information because it tries to present an immutable interface.
```

Comment:
I don't think this is why the checker gets created every time. The checker gets created every time because each program's type information is shared between the checker instance, and can be queried by each checker.